api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -598,6 +638,50 @@ private CohortDefinition getPatientsWithTransferFromOtherHF() {
     return cd;
   }
 
+  public CohortDefinition getAdditionalExclusionCriteriaForC1() {
+    CompositionCohortDefinition cd  = new CompositionCohortDefinition();
+    cd.setName("All patients to be excluded for the C1 definition");
+    cd.addParameter(new Parameter("onOrAfter", "onOrAfter", Date.class));
+    cd.addParameter(new Parameter("onOrBefore", "onOrBefore", Date.class));
+    cd.addParameter(new Parameter("locationList", "location", Location.class));
+    List<Concept> concepts = new ArrayList<>();
+    cd.addSearch("transferBasedOnObsDate", map(getPatientsTransferBasedOnObsDate(), "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},locationList=${locationList}"));
+    cd.addSearch("getTypeOfPatientTransferredFrom", map(getTypeOfPatientTransferredFrom(), "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},locationList=${locationList}"));
+    cd.addSearch("inProgramState", map(genericCohortQueries.getPatientsBasedOnPatientStates(hivMetadata.getHIVCareProgram().getProgramId(),
+            hivMetadata.getPateintTransferedFromOtherFacilityWorkflowState().getProgramWorkflowStateId()), "startDate=${onOrAfter},endDate=${onOrBefore},location=${locationList}"));
+    cd.setCompositionString("transferBasedOnObsDate OR getTypeOfPatientTransferredFrom OR inProgramState");
+    return cd;
+  }
+
+  public CohortDefinition getAdditionalExclusionCriteriaForC2() {
```

Comment:
Looks Like the **getAdditionalExclusionCriteriaForC2()** is similar to **getAdditionalExclusionCriteriaForC1()**, maybe you could create one function with a generic name and reuse it#

Reply:
The difference comes in the parameter manipulation, one needs one month be sutracted from startDate for both 2 searches and one just maintain. I would suggest those to stay that way for now